gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java

```diff
@@ -69,13 +145,21 @@ public void install() {
                     throw new IllegalStateException("Install can only be called once and should be called inside your application onCreate()");
                 }
 
+                if (this.context == null) {
+                    throw new IllegalStateException("You must supply a valid context");
+                }
+
                 if (StringUtils.isBlank(this.bootstrapUrl)) {
                     throw new IllegalStateException("You must supply a bootstrap url");
                 }
 
                 Gandalf gandalf = createInstance();
 
+                gandalf.context = this.context;
                 gandalf.bootstrapUrl = this.bootstrapUrl;
+                gandalf.versionChecker = new DefaultVersionChecker();
+                gandalf.historyChecker = new DefaultHistoryChecker(this.context);
+                gandalf.gateKeeper = new GateKeeper(this.context, gandalf.versionChecker, gandalf.historyChecker);
```

Comment:
Probably a good idea, I wish you could make something as final but have the compiler recognize a builder and allow that as well as a constructor.